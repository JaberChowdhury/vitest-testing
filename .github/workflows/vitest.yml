name: Run Vitest Tests

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      # Step 1: Check out the repository
      - name: Checkout code
        uses: actions/checkout@v3

      # Step 2: Set up Node.js with pnpm caching
      - name: Set up Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 18
          cache: "pnpm"

      # Step 3: Install pnpm manually (bypassing Corepack)
      - name: Install pnpm
        run: npm install -g pnpm

      # Step 4: Ensure pnpm is added to PATH
      - name: Add pnpm to PATH
        run: echo "$(npm root -g)/pnpm" >> $GITHUB_PATH

      # Step 5: Verify pnpm installation and check PATH
      - name: Verify pnpm installation
        run: |
          echo "pnpm version:"
          pnpm --version
          echo "pnpm location:"
          which pnpm
          echo "Current PATH:"
          echo $PATH

      # Step 6: Install dependencies
      - name: Install dependencies
        run: pnpm install

      # Step 7: Run Vitest tests
      - name: Run tests
        run: pnpm test
