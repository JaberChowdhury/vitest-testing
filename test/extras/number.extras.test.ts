import { describe, expect, test } from "vitest";

describe("number extras", () => {
  // Floating-Point Number Matching
  test("floating-point number match", () => {
    const num = 0.1 + 0.2; // 0.30000000000000004
    expect(num).toBeCloseTo(0.3, 5); // Second argument is the number of decimal places to check
  });
  //   Checking for NaN
  test("value is NaN", () => {
    const num = NaN;
    expect(num).toBeNaN();
  });
  //   Checking for Infinity
  test("value is Infinity", () => {
    const num = 1 / 0;
    expect(num).toBe(Infinity);
  });

  test("value is -Infinity", () => {
    const num = -1 / 0;
    expect(num).toBe(-Infinity);
  });
  //   Range Checking
  test("number is within range", () => {
    const num = 50;
    expect(num).toBeGreaterThan(0);
  });
  test("number is within range", () => {
    const num = 50;
    expect(num).toBeLessThan(100);
  });
  //   Checking for Integer
  test("number is an integer", () => {
    const num = 42;
    expect(Number.isInteger(num)).toBeTruthy();
  });
  // Snapshot Testing for Numbers
  test("number snapshot", () => {
    const num = 42;
    expect(num).toMatchSnapshot();
  });
});
