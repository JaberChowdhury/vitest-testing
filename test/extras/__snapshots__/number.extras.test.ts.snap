// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`number extras > number snapshot 1`] = `42`;
